### ISSUE TITLE
lottie: gradient slot issue

### ISSUE THREAD

**Issue by theashraf** [thorvg/thorvg#2795]
[gradient-fill.json](https://github.com/user-attachments/files/17199396/gradient-fill.json)

The provided Lottie file contains the following slots:

```json
"slots": {
  "gradient_fill": {
    "p": {
      "a": 0,
      "k": [0, 0.1, 0.1, 0.2, 1, 1, 0.1, 0.2, 0.1, 1]
    }
  }
}
```

I expected the gradient to render as shown in this image:
<img width="270" alt="Expected gradient" src="https://github.com/user-attachments/assets/bc51f10c-4366-4d72-856e-aa7bbc8f561e">

but, for some reason, it’s not being rendered correctly, as shown here:
<img width="257" alt="Incorrect gradient" src="https://github.com/user-attachments/assets/4b8eb8a6-89b8-430a-8e9f-de6d7f82f12b">

**Comment by tinyjin** [comment]
![CleanShot 2024-10-01 at 13 58 06@2x](https://github.com/user-attachments/assets/bf33d8ed-f5be-4ddf-b128-7c0e7ddbb9f1)

By replacing original gradient array, it renders what we expect. Something happens in overriding makes colorStops different.

I'll look into this. Thanks for reporting.

**Comment by tinyjin** [comment]
@theashraf 
I have a question regarding to overriden colorStop value.

Let's say original gradient has 2 colorStops in one sid before overriding.
Is new colorStop from slot same length as well?

Or any possible situation that different length of gradient will come and be overriden?

**Comment by mgrudzinska** [comment]
https://github.com/lottie/lottie-spec/issues/105

**Comment by hermet** [comment]
Even our sample don't have value count, shouldn't our Lottie Extension sample suggest like this?

![image](https://github.com/user-attachments/assets/7a7e8a86-7962-476a-acfb-54babdc455ac)

![image](https://github.com/user-attachments/assets/556d54ef-d034-467b-81a1-1ce3b85174d2)

**Comment by mgrudzinska** [comment]
yes, slotsample.json and test case lottie slot should be updated

**Comment by tinyjin** [comment]
agree, the "p" value would suggest how many colors stops will be changed

**Comment by hermet** [comment]
@theashraf Please review the form change carefully, leave any feedback. Otherwise, the fix will be in the next release (v0.15.4).

slot data to override:
```
{
    "gradient_fill": {
        "p": {
            "p": 2,      // 2 color stop count
            "k": {
                "k": [
                    0,
                    0.1,
                    0.1,
                    0.2,
                    1,
                    1,
                    0.1,
                    0.2,
                    0,
                    0,
                    1,
                    1
                ]
            }
        }
    }
}
```

slot data target:
```
"g": {
    "p": 9,   //9 color stop count
    "k": {
        "a": 0,
        "k": [
            0,
            0.514,
            0.373,
            0.984,
            0.141,
            0.478,
            0.412,
            0.984,
            0.283,
            0.443,
            0.451,
            0.984,
            0.379,
            0.408,
            0.49,
            0.984,
            0.475,
            0.373,
            0.529,
            0.984,
            0.606,
            0.278,
            0.647,
            0.925,
            0.737,
            0.184,
            0.765,
            0.867,
            0.868,
            0.092,
            0.882,
            0.808,
            1,
            0,
            1,
            0.749
        ]
    },
    "sid": "gradient_fill"
}
```

**Comment by theashraf** [comment]
@hermet looks good to me, i wonder if animated gradient slots is also supported ?
